bioptim/interfaces/SolverOptions.py

```diff
@@ -0,0 +1,95 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+
+@dataclass
+class SolverOptions(ABC):
+    @abstractmethod
+    def set_convergence_tol(self, tol):
```

Comment:
Please add the type of `tol` (`def set_convergence_tol(self, tol: float):`) for instance
(and carry to the subclasses)